app/components/HighlighterModule/highlighterModule.js

```diff
@@ -0,0 +1,437 @@
+var HighlighterModule = (function () {
+    var privateText = {},
+        privateTopics = [],
+        privateCurrentTopic = [],
+        privateDefColors = ['rgb(241, 96, 97)', 'rgb(253, 212, 132)', 'rgb(175, 215, 146)', 'rgb(168, 210, 191)', 'rgb(255,153,000)', 'rgb(102,000,153)', 'rgb(000,153,153)', 'rgb(255,102,255)', 'rgb(000,051,153)', 'rgb(153,000,204)', 'rgb(70,194,64)', 'rgb(94,242,188)'],
+        privateColors = [],
+        privateArticleIndex = [],
+        privateHighlights = [],
+        privateSelected = [],
+        privateContainer = 'no id specified';
+
+    /* ---------- Private Methods  --------- */
+
+    function mergeColors(topics, selected) {
+
+      var colors = privateColors;
+      var list = [];
+      var index = 0;
+      for (var current=0;current <topics.length;current+=1) {
+        for (var master=0;master <privateTopics.results.length;master+=1){
+          if (topics[current] == privateTopics.results[master].id) {
+            list.push(colors[master]);
+          }
+        }
+      }
+
+      var fraction = 1 / list.length;
+      var red = 0;
+      var blue = 0;
+      var green = 0;
+      index = 0;
+      while (index < list.length) {
```

Comment:
It's complex because originally the colors and topics weren't bundled together. If they were included together that would make life a whole lot easier.